### PR TITLE
[frontend] Fix update on attribute types

### LINKED ISSUE #5290: [production][attribute#update] undefined local variable or method `entry'
attribute#update threw this 4 times
<pre>
undefined local variable or method `entry&#39; for #&lt;AttributeController:0x00007f7dccee9520&gt;
</pre>
[https://api.opensuse.org/attribute/OSRT/Crawler/_meta](https://api.opensuse.org/attribute/OSRT/Crawler/_meta)"

[See this exception on Errbit](https://errbit-opensuse.herokuapp.com/apps/5620ca2bdc71fa00b1000000/problems/5b3b94e08d745000148dbfa8)

### PR DESCRIPTION
**Pull request by coolo**
POST needs to be done twice in tests - as :update is both
create and update in the attributes API

Fixes #5290

### CODE REVIEW

File: src/api/spec/controllers/attribute_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  render_views
+
+  describe '#update' do
+    let!(:admin) { create(:admin_user) }
+    let(:user1) { create(:confirmed_user) }
+    let!(:namespace) { create(:attrib_namespace, name: 'TEST') }
+    let(:attribute) { create(:attrib_type, attrib_namespace: namespace) }
+    let(:dummy) { 'Dummy' }
+
+    let(:data1) do
```
Comment (hennevogel) [inline src/api/spec/controllers/attribute_controller_spec.rb]: :thought_balloon: why not use a factory and `.to_xml` on it?

File: src/api/spec/controllers/attribute_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+RSpec.describe AttributeController, type: :controller do
+  render_views
+
+  describe '#update' do
+    let!(:admin) { create(:admin_user) }
+    let(:user1) { create(:confirmed_user) }
+    let!(:namespace) { create(:attrib_namespace, name: 'TEST') }
+    let(:attribute) { create(:attrib_type, attrib_namespace: namespace) }
+    let(:dummy) { 'Dummy' }
```
Comment (hennevogel) [inline src/api/spec/controllers/attribute_controller_spec.rb]: Huh?

File: src/api/spec/controllers/attribute_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      </default>
+      <allowed>
+        <value>A</value>
+        <value>B</value>
+      </allowed>
+      <modifiable_by user='#{user1.login}'/>
+    </definition>"
+    end
+
+    it 'will require auth' do
```
Comment (hennevogel) [inline src/api/spec/controllers/attribute_controller_spec.rb]: This is already tested in `spec/lib/authenticator_spec.rb`

File: src/api/spec/controllers/attribute_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      post :update, body: data1, format: :xml, params: { namespace: namespace.name, name: dummy }
+      expect(response.status).to eq(401)
+    end
+    it 'will require permissions' do
```
Comment (hennevogel) [inline src/api/spec/controllers/attribute_controller_spec.rb]: This is already tested in `app/policies/attrib_policy.rb`

File: src/api/spec/controllers/attribute_controller_spec.rb (hunk 1 above)
Comment (coolo) [reply src/api/spec/controllers/attribute_controller_spec.rb]: Hmm, I would I have to_xml without creating it?

File: src/api/spec/controllers/attribute_controller_spec.rb (hunk 2 above)
Comment (coolo) [reply src/api/spec/controllers/attribute_controller_spec.rb]: this is merely taken from the minitest

File: src/api/spec/controllers/attribute_controller_spec.rb (hunk 4 above)
Comment (coolo) [reply src/api/spec/controllers/attribute_controller_spec.rb]: ok, fair enough

File: src/api/spec/controllers/attribute_controller_spec.rb (hunk 2 above)
Comment (hennevogel) [reply src/api/spec/controllers/attribute_controller_spec.rb]: Ah okay, can you remove it anyway? :-)

File: src/api/spec/controllers/attribute_controller_spec.rb (hunk 1 above)
Comment (bgeuken) [reply src/api/spec/controllers/attribute_controller_spec.rb]: `build(:my_factory).to_xml` :wink:

Comment (bgeuken) [reply src/api/spec/controllers/attribute_controller_spec.rb]: And please using speaking names. What kind of data is this?

File: src/api/spec/controllers/attribute_controller_spec.rb (hunk 2 above)
Comment (coolo) [reply src/api/spec/controllers/attribute_controller_spec.rb]: no, as the variable (or whatever you call these lets) is used all over in the test :)

Comment (hennevogel) [reply src/api/spec/controllers/attribute_controller_spec.rb]: I thought you know `%:s`...

File: src/api/spec/controllers/attribute_controller_spec.rb (hunk 1 above)
Comment (coolo) [reply src/api/spec/controllers/attribute_controller_spec.rb]: there is no to_xml on attrib_types - so now?

**Comment by coolo** [comment]
simplified the test - please check.

**Comment by coolo** [comment]
ping

File: src/api/spec/controllers/attribute_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+RSpec.describe AttributeController, type: :controller do
+  render_views
+
+  describe '#update' do
+    let(:admin) { create(:admin_user) }
+    let(:user) { create(:confirmed_user) }
+    let(:attribute) { build(:attrib_type) }
+    let(:namespace) { attribute.namespace }
+    let(:name) { attribute.name }
```
Comment (hennevogel) [inline src/api/spec/controllers/attribute_controller_spec.rb]: :thought_balloon: why do this indirection for `name` and `namespace`?

File: src/api/spec/controllers/attribute_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+      login admin
+      post :update, body: xml_count_2, format: :xml, params: { namespace: namespace, name: name }
+      expect(response.status).to eq(200)
+
+      attrib = AttribType.find_by_namespace_and_name!(namespace, name)
+      expect(attrib).not_to be_nil
+      expect(attrib.namespace).to eq(namespace)
+    end
+
+    it 'will update on POST' do
```
Comment (hennevogel) [inline src/api/spec/controllers/attribute_controller_spec.rb]: I don't like that you create test data by sending it. Why don't you use the factories to create the `AttribType` with `count=2` that you update in the second step?

File: src/api/spec/controllers/attribute_controller_spec.rb (hunk 5 above)
Comment (coolo) [reply src/api/spec/controllers/attribute_controller_spec.rb]: to save typing?

File: src/api/spec/controllers/attribute_controller_spec.rb (hunk 6 above)
Comment (coolo) [reply src/api/spec/controllers/attribute_controller_spec.rb]: because I want to test posting XML, not rendering models.

Comment (hennevogel) [reply src/api/spec/controllers/attribute_controller_spec.rb]: and testing posting XML twice is better or what?

Comment (coolo) [reply src/api/spec/controllers/attribute_controller_spec.rb]: well, you're free to write the spec yourself - I wanted to fix a bug and write a test case for it. Not to win a coding beauty contest.

Comment (hennevogel) [reply src/api/spec/controllers/attribute_controller_spec.rb]: Sorry but we have some coding standards here. This is not about being beautiful but about clarity for the people who have to maintain this...
```
<norddeutsch>
Sei nich so unegal man...
</norddeutsch>
```

Comment (coolo) [reply src/api/spec/controllers/attribute_controller_spec.rb]: ¿mejor ahora?

**Comment by hennevogel** [comment]
No idea about the travis/circle difference...

**Comment by coolo** [comment]
I don't trust future developers not to break the test by putting a ! behind the let

**Comment by coolo** [comment]
codecov has no base report

**Pull request merged**